src/orange/orangeinp/CsgTree.cc

```diff
@@ -166,6 +166,65 @@ auto CsgTree::simplify(NodeId node_id) -> Simplification
     return Simplification{std::move(repl)};
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Simplify negated joins using De Morgan's law.
+ *
+ * This is required if the tree's logic expression is used with
+ * \c InfixEvaluator as negated joins are not supported.
+ */
+void CsgTree::simplify_negated_joins()
```

Comment:
Is it possible to make this an external `CsgTreeUtils` function rather than extending the class?